Sources/SwiftBluetooth/Peripheral/Peripheral+DiscoveryInfo.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+import CoreBluetooth
+
+extension Peripheral {
+    public struct DiscoveryInfo {
+        public var RSSI: Int
+        public var advertisementData: AdvertisementData
+
+        init(RSSI: NSNumber, advertisementData: [String : Any]) {
+            self.RSSI = Int(truncating: RSSI)
+            self.advertisementData = .init(data: advertisementData)
+        }
+
+        public struct AdvertisementData {
+            private(set) var data: [String : Any]
+
+            subscript(key: String) -> Any? {
+                data[key]
+            }
+
+            public var localName: String? {
+                data[CBAdvertisementDataLocalNameKey] as? String
+            }
+
+            public var manufacturerData: Data? {
+                data[CBAdvertisementDataManufacturerDataKey] as? Data
+            }
+
+            public var serviceData: [CBUUID : Data]? {
```

Comment:
Optional dictionaries and arrays are generally an anti-pattern in Swift. It's better to just use an empty dictionary instead of `nil`.

Reply:
Determining if the key was advertised at all might be of use to some people. Though for ease-of-use I probably agree that adding a default here is much more convenient.

I guess they can always index on the underlying CoreBluetooth dictionary to determine that though. Will implement (for the arrays too).